rocketpool-cli/service/service.go

```diff
@@ -1099,8 +1099,92 @@ func pruneExecutionClient(c *cli.Context) error {
 
 }
 
-// Pause the Rocket Pool service
-func pauseService(c *cli.Context) error {
+// Stops Smartnode stack containers, prunes docker, and restarts the Smartnode stack.
+func resetDocker(c *cli.Context) error {
+
+	fmt.Println("Once cleanup is complete, Rocket Pool will restart automatically.")
+	fmt.Println()
+
+	// Stop...
+	// NOTE: pauseService prompts for confirmation, so we don't need to do it here
+	confirmed, err := pauseService(c)
+	if err != nil {
+		return err
+	} else if !confirmed {
+		// if the user cancelled the pause, then we cancel the rest of the operation here:
+		return nil
+	}
+
+	// Prune images...
+	err = pruneDocker(c)
+	if err != nil {
+		return fmt.Errorf("error pruning Docker: %s", err)
+	}
+
+	// Restart...
+	// NOTE: startService does some other sanity checks and messages that we leverage here:
+	fmt.Println("Restarting Rocket Pool...")
+	err = startService(c, true)
+	if err != nil {
+		return fmt.Errorf("error starting Rocket Pool: %s", err)
+	}
+	return nil
+}
+
+func pruneDocker(c *cli.Context) error {
+
+	// Get RP client
+	rp := rocketpool.NewClientFromCtx(c)
+	defer rp.Close()
+
+	// NOTE: we deliberately avoid using `docker system prune -a` and delete all
+	//   images manually so that we can preserve the current smartnode-stack
+	//   images, _unless_ the user specified --all option
+	deleteAllImages := c.Bool("all")
+	if !deleteAllImages {
+		ourImages, err := rp.GetComposeImages(getComposeFiles(c))
+		if err != nil {
+			return fmt.Errorf("error getting compose images: %w", err)
+		}
+
+		ourImagesMap := make(map[string]struct{})
+		for _, image := range ourImages {
+			ourImagesMap[image] = struct{}{}
+		}
+
+		allImages, err := rp.GetAllDockerImages()
+		if err != nil {
+			return fmt.Errorf("error getting all docker images: %w", err)
+		}
+
+		fmt.Println("Deleting images not used by the Rocket Pool Smartnode...")
+		for _, image := range allImages {
+			if _, ok := ourImagesMap[image.TagString()]; !ok {
+				fmt.Printf("Deleting %s...\n", image.String())
+				_, err = rp.DeleteDockerImage(image.ID)
+				if err != nil {
+					// safe to ignore and print to user, since it may just be an image referenced by a running container that is managed outside of the smartnode's compose stack
+					fmt.Printf("Error deleting image %s: %s\n", image.String(), err.Error())
+				}
+			} else {
+				fmt.Printf("Skipping image used by Smartnode stack: %s\n", image.String())
+			}
```

Comment:
```suggestion
				}
				continue
			}
			
			fmt.Printf("Skipping image used by Smartnode stack: %s\n", image.String())
```

keep the happy path at the primary indentation level of the loop scope